### ISSUE TITLE
Review version generation

### ISSUE THREAD

**Issue by rsmmr** [zeek/spicy#956]
Let's review our we generate version numbers. I keep seeing inconsistencies. 

Most recent example: `brew install spicy —HEAD` installs a `spicyc` that reports itself as `spicyc v1.1.0-branch (701e5137)`, even though that hash corresponds to `1.2.0-dev.87`. Also, sometimes I end up having `VERSION` files even when working from git (which, iirc, then take precendence over git).

Also, a fresh Spicy `main` clone shows in the configure output:

```
====================|  Spicy Build Summary  |====================

Version:               1.2.0
[...]
```

We should then also update minimum Spicy version numbers for the plugin and analyzer repos.

**Comment by rsmmr** [comment]
To expand, both `spicy-config -v` and that `Version: ...` output should be showing the current `X.Y.Z-dev-NN` version number

**Comment by rsmmr** [comment]
Also, when I'm on the `release/1.1` branch, I get this:

```
# ./scripts/autogen-version
1.2.0-dev.13 (release/1.1 26e5c6b1)
```

**Comment by bbannier** [comment]
I looked into this.

> `brew install spicy —HEAD` installs a `spicyc` that reports itself as `spicyc v1.1.0-branch (701e5137)`, even though that hash corresponds to `1.2.0-dev.87`.

I was unable to reproduce this. Currently the Spicy Homebrew formula performs a deep clone of the `main` branch and I cannot see how this would happen. I also validated that this isn't caused by lingering `VERSION` files in the source directory; Homebrew seems to cache a clone, and then perform another ephemeral clone of that cache when performing the build, so the `VERSION` file should not persist across multiple installs.

> Also, when I'm on the release/1.1 branch, ...

I was unable to reproduce this, but since the tag `1.2.0-dev` is an ancestor of the branch `release/1.1` it seems at least possible for the `git describe` we use in `scripts/autogen-version` to pick an older tag as better candidate, but like I wrote, this does not repro for me.

We can probably avoid this kind of issue by branching a release branch first, and making the placeholder `X.Y.Z-dev` commit and tag on the `main` branch so it is not in the release branch's history (I clarified the release checklist for that).

> Also, sometimes I end up having VERSION files even when working from git (which, iirc, then take precendence over git).

We always emit a `VERSION` file during the configure phase. This file will by design take precedence, so I am not sure what we can do in our current framework.

A possible different approach would be to explicitly track a version in some `VERSION` file in version control (e.g., on `main` `1.2.0-dev`, on `release/1.1` `1.1.1-dev`, ...). During the configure phase we would only extract the Git commit for our long version string (not sure whether branch information is crucial). In that model we'd avoid problems relating where tool old tags end up in the final version.

> To expand, both spicy-config -v and that Version: ... output should be showing the current X.Y.Z-dev-NN version number

I think having commit information in both the output of `<prog> -v` and the CMake summary could make debugging issues easier. How about us instead emitting a long version string (the one we currently emit from the programs) in the CMake summary as well? I will open a PR for that.

**Comment by bbannier** [comment]
I merged a change which includes additional changes to make sure that the generated `VERSION` file is up-to-date.

**Comment by rsmmr** [comment]
So with 1.2.0, I'm now seeing these versions strings:

```
# brew install spicy
# spicy -v 
spicyc v1.2.0 (f3ba770e f3ba770e)

# brew install spicy —HEAD
# spicy -v 
spicyc v1.2.0.2 (main d9addab5)
```

The 1st looks good, could just use some polishing to show the commit only once. But the 2nd seems odd.

**Comment by bbannier** [comment]
As a follow up I changed to the approach of directly maintaining the `VERSION` file in the repo, #965.

**Comment by bbannier** [comment]
Closing this as the fix from #965 seems to work fine.
